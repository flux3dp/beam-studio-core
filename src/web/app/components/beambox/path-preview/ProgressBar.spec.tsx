import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ProgressBar from './ProgressBar';

test('should render correctly', () => {
  const handleSimTimeChange = jest.fn();
  const wrapper = shallow(<ProgressBar
    simTime={0.5}
    simTimeMax={1}
    handleSimTimeChange={handleSimTimeChange}
  />);
  expect(toJson(wrapper)).toMatchSnapshot();

  wrapper.find('input.slider').simulate('change', {
    target: { value: 0.7 },
  });
  expect(handleSimTimeChange).toHaveBeenCalledTimes(1);
  expect(handleSimTimeChange).toHaveBeenNthCalledWith(1, 0.000011666666666666666);
});
