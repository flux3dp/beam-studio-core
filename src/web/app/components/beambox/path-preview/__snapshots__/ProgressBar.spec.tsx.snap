// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    className="label pull-left"
  />
  <div
    className="path-preview-slider-container"
    id="progress-bar"
  >
    <input
      className="slider"
      max={60000}
      min={0}
      onChange={[Function]}
      step={100}
      style={
        Object {
          "--percentage": "50%",
        }
      }
      type="range"
      value={30000}
    />
  </div>
</div>
`;
